dev/com.ibm.ws.sipcontainer/src/com/ibm/ws/sip/channel/resolver/impl/netty/SipResolverTcpTransport.java

```diff
@@ -0,0 +1,754 @@
+
+        /** open the listener socket */
+        try {
+            _framework.startOutbound(bootstrap, _currentSocketAddress.getHostString(), _currentSocketAddress.getPort(),
+                    f -> {
+                        if (f.isCancelled() || !f.isSuccess()) {
+                            if (c_logger.isWarnEnabled()) {
+                                c_logger.warn("Resolver channel exception during connect: " + f.cause().getMessage());
+                            }
+                            destroy((Exception) f.cause());
+                        } else {
+                            m_channel = f.channel();
+                            ready();
+                        }
+                    });
+        } catch (NettyException e) {
+            e.printStackTrace();
+            if (c_logger.isWarnEnabled()) {
+                c_logger.warn("Resolver channel exception during connect: " + e.getMessage());
+            }
+        }
+
+//        final ChannelFuture channelFuture = bootstrap.connect(_currentSocketAddress);
+//
+//        channelFuture.addListener(f -> {
+//            if (f.isCancelled() || !f.isSuccess()) {
+//                if (c_logger.isWarnEnabled()) {
+//                    c_logger.warn("Resolver channel exception during connect: " + f.cause().getMessage());
+//                }
+//                destroy((Exception) f.cause());
+//            } else {
+//                m_channel = channelFuture.channel();
+//                ready();
+//            }
+//        });
+        // Connection established successfully
+
+        if (c_logger.isTraceEntryExitEnabled())
+            c_logger.traceExit(this, "SipResolverTcpTransport: connect: exit: id=" + hashCode());
+    }
+
+    /**
+     * 
+     */
+    synchronized public void writeRequest(ByteBuf requestBuffer) throws IOException {
+        if (c_logger.isTraceEntryExitEnabled())
+            c_logger.traceEntry(this, "SipResolverTcpTransport: writeRequest: entry: id=" + hashCode());
+
+        if (_shutdown == true)
+            throw new IllegalStateException("SIP TCP Resolver transport is shutdown.");
+
+        switch (_writeState) {
+        case WRITE_STATE_SHUTDOWN:
+            // unreachable? debug only
+            if (c_logger.isTraceDebugEnabled())
+                c_logger.traceDebug("SipResolverTcpTransport:writeRequest: WRITE_STATE_SHUTDOWN");
+            break;
+
+        case WRITE_STATE_IDLE:
+
+            if (c_logger.isTraceDebugEnabled())
+                c_logger.traceDebug("SipResolverTcpTransport:writeRequest: WRITE_STATE_IDLE");
+
+            _lengthBuffer.clear();
+            _lengthBuffer.writeShort((short) requestBuffer.readableBytes());
+
+//            System.out.println("writeRequest writableBytes" + requestBuffer.readableBytes());
+
+            _bufferArray[0] = _lengthBuffer;
+            _bufferArray[1] = requestBuffer;
+            ByteBuf newBuf = Unpooled.copiedBuffer(_bufferArray);
```

Comment:
Bill, was there a glitch with the composite buffer below that was solved by using an unpooled copied buffer above?